src/main/java/org/rumbledb/server/RumbleHandler.java

```diff
@@ -0,0 +1,145 @@
+package org.rumbledb.server;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+import java.net.URLDecoder;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.spark.SparkException;
+import org.rumbledb.api.Item;
+import org.rumbledb.cli.JsoniqQueryExecutor;
+import org.rumbledb.config.RumbleRuntimeConfiguration;
+import org.rumbledb.exceptions.OurBadException;
+import org.rumbledb.exceptions.SparksoniqRuntimeException;
+import org.rumbledb.items.ItemFactory;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+
+import sparksoniq.spark.SparkSessionManager;
+
+public class RumbleHandler implements HttpHandler {
+
+    public RumbleHandler() {
+    }
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+        try {
+            URI uri = exchange.getRequestURI();
+            String query = uri.getQuery();
+            Map<String, String> queryParameters = new HashMap<String, String>();
+            for (String pair : query.split("&")) {
+                int index = pair.indexOf("=");
+                queryParameters.put(
+                    URLDecoder.decode(pair.substring(0, index), "UTF-8"),
+                    URLDecoder.decode(pair.substring(index + 1), "UTF-8")
+                );
+            }
+            String[] args = new String[queryParameters.keySet().size() * 2];
+            int i = 0;
+            for (String param : queryParameters.keySet()) {
+                args[i++] = "--" + param;
+                args[i++] = queryParameters.get(param);
+            }
+            RumbleRuntimeConfiguration configuration = new RumbleRuntimeConfiguration(args);
+            SparkSessionManager.COLLECT_ITEM_LIMIT = configuration.getResultSizeCap();
+
+            JsoniqQueryExecutor translator = new JsoniqQueryExecutor(configuration);
+            List<Item> items = translator.runQuery(
+                configuration.getQueryPath(),
+                configuration.getOutputPath()
+            );
+
+            Item output = ItemFactory.getInstance().createObjectItem();
+            if (items != null) {
+                Item values = ItemFactory.getInstance().createArrayItem(items);
+                output.putItemByKey("values", values);
+            }
+            output.putItemByKey("status", ItemFactory.getInstance().createIntegerItem(200));
```

Comment:
I would like to recommend extracting the HTTP status codes to static variables. While 200 and 500 are well known, if we add more codes in the future, these magic values may become obfuscated.

Reply:
DONE, thanks! Have you tested it?

    $spark-submit target/spark-rumble-1.6.0-jar-with-dependencies.jar --server yes
    $curl --data "parallelize(1 to 10)[2]" -X POST "http://localhost:8000/jsoniq"

Reply:
Just tested it. Works beautifully 👍  Nicely done

Reply:
`Line 126: ... createIntegerItem(200)`

Can we use the StatusCode Enum here as well?